crates/ra_hir_def/src/nameres/raw.rs

```diff
@@ -401,14 +402,28 @@ impl RawItemsCollector {
 
         let name = m.name().map(|it| it.as_name());
         let ast_id = self.source_ast_id_map.ast_id(&m);
-        // FIXME: cfg_attr
-        let export = m.attrs().filter_map(|x| x.simple_name()).any(|name| name == "macro_export");
 
         // FIXME: cfg_attr
-        let builtin =
-            m.attrs().filter_map(|x| x.simple_name()).any(|name| name == "rustc_builtin_macro");
-
-        let m = self.raw_items.macros.alloc(MacroData { ast_id, path, name, export, builtin });
+        let export = attrs.by_key("macro_export").exists();
+        let local_inner =
+            attrs.by_key("macro_export").tt_values().map(|it| &it.token_trees).flatten().any(
```

Comment:
We are looking up the `macro_export` twice here. I'm not sure how fast the lookup is, but maybe the code would be cleaner if we only looked it up once and handled appropriate cases based on the lookup.

Reply:
Yes, the problem here is [`AttrQuery`][1] take `self` , and I can't reuse the query,  I might make it copyable ..

[1]: https://github.com/rust-analyzer/rust-analyzer/blob/67afeebaf597765828801d7cddcb2976a0ee135d/crates/ra_hir_def/src/attr.rs#L148

Reply:
Done !